src/recording/recordingmanager.cpp

```diff
@@ -145,23 +167,37 @@ QString& RecordingManager::getRecordingDir() {
 }
 
 // Only called when recording is active.
-void RecordingManager::slotDurationRecorded(QString durationStr)
+void RecordingManager::slotDurationRecorded(quint64 duration)
 {
-    if(m_durationRecorded != durationStr)
+    if(m_secondsRecordedSplit != duration)
     {
-        m_durationRecorded = durationStr;
-        emit(durationRecorded(m_durationRecorded));
+        m_secondsRecordedSplit = duration;
+        if(duration >= m_split_time)
+        {
+            qDebug() << "Splitting after " << duration << " seconds";
+            // Dont generate a new filename.
+            // This will reuse the previous filename but append a suffix.
+            startRecording(false);
+        }
+        emit(durationRecorded(getRecordedDurationStr(m_secondsRecorded+m_secondsRecordedSplit)));
     }
 }
+// Copy from the implementation in enginerecord.cpp
+QString RecordingManager::getRecordedDurationStr(quint64 duration) {
+    return QString("%1:%2")
+                 .arg(duration / 60, 2, 'f', 0, '0')   // minutes
+                 .arg(duration % 60, 2, 'f', 0, '0');  // seconds
+}
 
 // Only called when recording is active.
 void RecordingManager::slotBytesRecorded(int bytes)
 {
     // auto conversion to long
     m_iNumberOfBytesRecorded += bytes;
-    if(m_iNumberOfBytesRecorded >= m_split_size)
+    m_iNumberOfBytesRecordedSplit += bytes;
+    if(m_iNumberOfBytesRecordedSplit >= m_split_size)
```

Comment:
I don't understand that sentence. I will explain what I do just in case:

enginerecord emits a signal with the number of bytes processed (i.e. the size of the buffer that the process call worked on).
on recordingmanager, we collect that size and increment a variable, so that we can compare how big the file is (in order to split it), but also in order to emit a signal to the UI (recordings, in the library), where it shows the total amount recorded.

The implementation as it was, needs the amount to be reset once we reach split_size, because we create a new file and start from 0, but then, we were telling to the UI that we started again too.
With these separated variables, we keep the current file size in one variable, and the total size recorded in the other.